tests/VerifyEmailTestKernel.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the SymfonyCasts VerifyEmailBundle package.
+ * Copyright (c) SymfonyCasts <https://symfonycasts.com/>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SymfonyCasts\Bundle\VerifyEmail\Tests;
+
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use SymfonyCasts\Bundle\VerifyEmail\SymfonyCastsVerifyEmailBundle;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+class VerifyEmailTestKernel extends Kernel
+{
+    private $builder;
+    private $routes;
+    private $extraBundles;
+
+    /**
+     * @param array $routes  Routes to be added to the container e.g. ['name' => 'path']
+     * @param array $bundles Array of bundles to be registered e.g. [new Bundle()]
+     */
+    public function __construct(ContainerBuilder $builder = null, array $routes = [], array $bundles = [])
+    {
+        $this->builder = $builder;
+        $this->routes = $routes;
+        $this->extraBundles = $bundles;
+
+        parent::__construct('test', true);
+    }
+
+    public function registerBundles(): iterable
+    {
+        return array_merge(
+            $this->extraBundles,
+            [
+                new FrameworkBundle(),
+                new SymfonyCastsVerifyEmailBundle(),
+            ]
+        );
+    }
+
+    public function registerContainerConfiguration(LoaderInterface $loader): void
+    {
+        if (null === $this->builder) {
+            $this->builder = new ContainerBuilder();
+        }
+
+        $builder = $this->builder;
+
+        $loader->load(function (ContainerBuilder $container) use ($builder) {
+            $container->merge($builder);
+            $container->loadFromExtension(
+                'framework',
+                [
+                    'secret' => 'foo',
+                    'router' => [
+                        'resource' => 'kernel::loadRoutes',
+                        'type' => 'service',
+                        'utf8' => true,
+                    ],
+                ]
+            );
+
+            $container->addObjectResource($this);
+            $container->register('kernel', static::class)
+                ->addTag('controller.service_arguments')
+                ->setAutoconfigured(true)
+                ->setSynthetic(true)
+                ->setPublic(true)
+            ;
```

Comment:
I don't think you need this section - almost definitely not the `->addTag()` part, but maybe also ALL of these lines can be removed. We're just not making use of them.